File: hw/collaguazo/u3/hw28Principles/Computers/controller/ComputerController.py
class ComputerController:
    def __init__(self, model):
        self.model = model
        self.fine_rate = 0.03 

    def save_registry(self, brand, rental_fee, days):
        try:
            daily_fee = float(rental_fee)
            delay_days = int(days)

            base_cost = daily_fee * delay_days
            fine_amount = round(base_cost * self.fine_rate, 2)
            total_pay = round(base_cost + fine_amount, 2)

            data = {
                "brand": brand,
                "daily_fee": daily_fee,
                "delay_days": delay_days,
                "tax": fine_amount,
                "total": total_pay
            }

            return self.model.insert(data)

        except Exception as e:
            print(f"Controller Error: {e}")
            return None

    def list_computers(self):
        return self.model.get_all()

File: hw/collaguazo/u3/hw28Principles/Library/src/main/java/ec/edu/espe/computer/controller/ComputerController.java
package ec.edu.espe.computer.controller;

import ec.edu.espe.computer.model.ILibrary;
import java.util.List;
import org.bson.Document;

/**
 *
 * @author Pablo Collaguazo
 */
public class ComputerController {

    private final ILibrary model;
    private final double fineRate = 0.03; 

    public ComputerController(ILibrary model) {
        this.model = model; 
    }

    public boolean saveRegistry(String title, double fee, int days) {
        try {
            double baseCost = fee * days;
            double fineAmount = Math.round((baseCost * fineRate) * 100.0) / 100.0;
            double totalPay = Math.round((baseCost + fineAmount) * 100.0) / 100.0;

            Document data = new Document("title", title)
                    .append("daily_fee", fee)
                    .append("delay_days", days)
                    .append("tax", fineAmount)
                    .append("total", totalPay);

            model.insert(data);
            return true;
        } catch (Exception e) {
            System.err.println("Error " + e.getMessage());
            return false;
        }
    }

    public List<Document> listComputers() {
        return model.getAll();
    }

}
